src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2499,6 +2506,13 @@ private bool ReuseOldCaches(string[] inputCacheFiles)
             }
         }
 
+        private void LogMessage(string message)
+        {
+            var loggingService = ((IBuildComponentHost)this).LoggingService;
+
+            loggingService?.LogCommentFromText(BuildEventContext.Invalid, MessageImportance.High, message);
```

Comment:
Should we create a new BuildEventArg type for logs generated by the engine (by the Scheduler, BuildManager, BuildRequestEngine, etc)?

Reply:
I'd really like to have BuildEventArgs structured differently so we can avoid problems like [this](https://github.com/microsoft/msbuild/issues/5019). That being said, we already have a lot, and adding a new one would make this logging a bit cleaner...just also make changing them all to use a visitor pattern a little harder.

Reply:
I like more-specific events but am fine punting this. I think it mostly depends on if you want it in a logger in BuildXL or another application.

Reply:
BXL and QB would use the API to get the information, not loggers. Let's leave it like this and upgrade to typed event if need to.